tools/metadata/webfeatures/schema.py

```diff
@@ -10,17 +11,36 @@
 """
 WEB_FEATURES_YML_FILENAME = "WEB_FEATURES.yml"
 
+# File prefix to indicate that this FeatureFile should run in EXCLUDE mode.
+EXCLUSION_PREFIX = "!"
+
 
 class SpecialFileEnum(Enum):
     """All files recursively"""
     RECURSIVE = "**"
 
 
+class FileMatchingMode(Enum):
+    """Defines how a FeatureFile pattern is used for matching."""
+    INCLUDE = 1  # Include files that match the pattern
+    EXCLUDE = 2  # Exclude files that match the pattern
+
 class FeatureFile(str):
+    @cached_property
+    def matching_mode(self) -> FileMatchingMode:
+        """Determines if the pattern should include or exclude matches."""
+        return FileMatchingMode.EXCLUDE if self.startswith(EXCLUSION_PREFIX) else FileMatchingMode.INCLUDE
+
+    @cached_property
+    def processed_filename(self) -> str:
+        """Removes the exclusion prefix "!" from the pattern."""
+        # TODO. After moving to Python3.9, use: return self.removeprefix(EXCLUSION_PREFIX)
+        return self[self.startswith(EXCLUSION_PREFIX) and len(EXCLUSION_PREFIX):]
```

Comment:
I'm haven't seen this style before. One branch is equivalent to `self[False:]` which in turn is equivalent to `self[0:]` and just `self`. It works, but I wonder if you find this more readable even though it's a bit longer:

```suggestion
        return self[len(EXCLUSION_PREFIX):] if self.startswith(EXCLUSION_PREFIX) else self
```